pkg/k8sutil/resource.go

```diff
@@ -82,6 +95,18 @@ func Reconcile(log zap.Logger, client runtimeClient.Client, desired runtimeClien
 				svc.ResourceVersion = current.(*corev1.Service).ResourceVersion
 				svc.Spec.ClusterIP = current.(*corev1.Service).Spec.ClusterIP
 				desired = svc
+			case *certv1.ClusterIssuer:
```

Comment:
If i'm not mistaken, users can provide custom issuers ([ref](https://github.com/konpyutaika/nifikop/blob/master/api/v1/nificluster_types.go#L428)) that don't belong to cert-manager. What happens if a custom provided issuer ref doesn't hit these cases?

Reply:
Is it true? Won't this function only be called with : 
```golang
// reconcileIssuer ensures a cert-manager Issuer.
func reconcileIssuer(ctx context.Context, log zap.Logger, client client.Client, issuer *certv1.Issuer, cluster *v1.NifiCluster) error {
	obj := &certv1.Issuer{}
	var err error
	if err = client.Get(ctx, types.NamespacedName{Name: issuer.Name, Namespace: issuer.Namespace}, obj); err != nil {
		if !apierrors.IsNotFound(err) {
			return err
		}
		return client.Create(ctx, issuer)
	}
	if err = k8sutil.Reconcile(log, client, issuer, cluster, &cluster.Status); err != nil {
		return err
	}
	return nil
}
```

I don't see in which other occasions it will be called, do you see one?

Reply:
Okay, asked a different way -- would these changes impact using a custom issuer? We use a non-cert-manager issuer with a custom CA.

Reply:
Like that I don't see how, but maybe I'm missing something 😕